unit_test/test_spdm_responder/key_pair_info.c

```diff
@@ -0,0 +1,319 @@
+                     public_key_info_len,
+                     public_key_info_rsa2048,
+                     public_key_info_len);
+
+    response_size = sizeof(response);
+
+    uint32_t session_id;
+    libspdm_session_info_t *session_info;
+
+    session_id = 0xFFFFFFFF;
+    spdm_context->latest_session_id = session_id;
+    spdm_context->last_spdm_request_session_id_valid = true;
+    spdm_context->last_spdm_request_session_id = session_id;
+    session_info = &spdm_context->session_info[0];
+    libspdm_session_info_init(spdm_context, session_info, session_id, true);
+    libspdm_secured_message_set_session_state(
+        session_info->secured_message_context,
+        LIBSPDM_SESSION_STATE_ESTABLISHED);
+
+    status = libspdm_get_response_key_pair_info(
+        spdm_context, m_libspdm_get_key_pair_info_request1_size,
+        &m_libspdm_get_key_pair_info_request1, &response_size, response);
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(response_size,
+                     sizeof(spdm_key_pair_info_response_t) + public_key_info_len);
+    spdm_response = (void *)response;
+    assert_int_equal(spdm_response->header.request_response_code,
+                     SPDM_KEY_PAIR_INFO);
+    assert_int_equal(spdm_response->total_key_pairs, 1);
+    assert_int_equal(spdm_response->key_pair_id, key_pair_id);
+    assert_int_equal(spdm_response->capabilities, SPDM_KEY_PAIR_CAP_GEN_KEY_CAP);
+    assert_int_equal(spdm_response->key_usage_capabilities, SPDM_KEY_USAGE_BIT_MASK_KEY_EX_USE);
+    assert_int_equal(spdm_response->current_key_usage, SPDM_KEY_USAGE_BIT_MASK_KEY_EX_USE);
+    assert_int_equal(spdm_response->asym_algo_capabilities, SPDM_KEY_PAIR_ASYM_ALGO_CAP_RSA2048);
+    assert_int_equal(spdm_response->public_key_info_len, sizeof(public_key_info_rsa2048));
+    assert_int_equal(spdm_response->assoc_cert_slot_mask, 0x01);
+    assert_memory_equal(spdm_response + 1, public_key_info_rsa2048,
+                        spdm_response->public_key_info_len);
+
+}
+libspdm_test_context_t m_libspdm_responder_key_pair_info_test_context = {
+    LIBSPDM_TEST_CONTEXT_VERSION,
+    false,
+};
+
+int libspdm_responder_key_pair_info_test_main(void)
+{
+    const struct CMUnitTest spdm_responder_key_pair_info_tests[] = {
+        /* Success Case to get key pair info*/
+        cmocka_unit_test(libspdm_test_responder_key_pair_info_case1),
+        /* An SPDM endpoint shall contain KeyPairID s starting from 1 to TotalKeyPairs inclusive and without gaps. */
+        cmocka_unit_test(libspdm_test_responder_key_pair_info_case2),
+        /* key_pair_id > total_key_pairs*/
+        cmocka_unit_test(libspdm_test_responder_key_pair_info_case3),
+        /* capability not set KEY_PAIR_INFO*/
+        cmocka_unit_test(libspdm_test_responder_key_pair_info_case4),
+        /* last_spdm_request_session_id_valid true */
+        cmocka_unit_test(libspdm_test_responder_key_pair_info_case5),
```

Comment:
`last_spdm_request_session_id_valid true`
case5 is get_key_pair_info in session, right?